bpfd/src/command.rs

```diff
@@ -191,15 +192,17 @@ pub(crate) struct TracepointProgram {
 #[derive(Serialize, Deserialize, Clone)]
 pub(crate) struct ProgramData {
     pub(crate) location: String,
-    pub(crate) path: String,
     pub(crate) section_name: String,
+    pub(crate) global_data: HashMap<String, Vec<u8>>,
```

Comment:
Hrm should this be an `Option<HashMap<String, Vec<u8>>` ?

Also I think we should change the new_from_location `ProgramData` function to just be `new` now that we're parsing global data and a location

Reply:
> Hrm should this be an `Option<HashMap<String, Vec<u8>>` ?

I tried defining it as `optional` in `bpfd.proto`, but I get the following error:
```
protoc failed: bpfd.proto:40:21: Field labels (required/optional/repeated) are not allowed on map fields.
```
So, I need to include an empty hashmap in `LoadRequest` if the user doesn't enter anything.  Also, @dave-tucker said in the issue:
> This may not need to be optional, since an empty map will be fine.

So, I just carried that through to bpfd.  Also, the code is simpler that way.  I can just do
```rust
for (name, value) in &program.data.global_data {
    loader.set_global(name, value.as_slice());
}
```
Instead of wrapping that same thing in an `if let`. 
However, it could be an `Option` in the `ProgramData` struct if we prefer.

Reply:
> Also I think we should change the new_from_location ProgramData function to just be new now that we're parsing global data and a location

I thought that was a weird name since it's the only `new` method we have and we're providing more than just the location.  I didn't think adding another parameter changed it, but I'm happy to change this to just `new`.

Reply:
> but I'm happy to change this to just new

I think that'd be good! 

Ok that all sounds good for now, that last thing I'm wondering is does the "save" functionality work correctly for a Hashmap i.e if bpfd restarts will it still correctly come back up? (I don't think we have any testing on this)

Reply:
Yes, save and load work correctly for hashmaps.  However, we don't reload the programs after a bpfd restart, so it doesn't get used.